frontend/src/metabase/query_builder/components/DataSelector.jsx

```diff
@@ -429,10 +438,24 @@ export default class DataSelector extends Component {
   previousStep = () => {
     const previousStep = this.getPreviousStep();
     if (previousStep) {
-      this.switchToStep(previousStep, {}, false);
+      const clearedState = this.getClearedStateForStep(previousStep);
+      this.switchToStep(previousStep, clearedState, false);
     }
   };
 
+  getClearedStateForStep(step) {
+    // TODO: do we need to clear state for other steps?
```

Comment:
@tlrobinson would uncleared state be problematic for any other steps? I should probably add similar clearing code just for consistency, but it'd be nice to add some failing tests too.